### PR TITLE
Changes for Inotable repair during Forward scrub 

### PR DESCRIPTION
**Pull request by vishalkanaujia**

### CODE REVIEW

File: src/mds/CInode.cc [hunk 1]
```diff
@@ -3837,6 +3839,25 @@ void CInode::validate_disk_state(CInode::validated_data *results,
...
+	      std::stringstream ds;
+	      inode_t& inode = in->inode;
+	      JSONFormatter *f = new JSONFormatter();
+              MDCache *mdcache = in->mdcache;
+
+	      mdcache->mds->inotable->dump(f);
+	      f->flush(ds);
+	      dout(0) << "scrub: inotable: adding ino = " << inode.ino <<dendl;
+	      dout(0) << "scrub: inotable status: before = "<< ds.str() << dendl;
+	      mdcache->mds->inotable->repair_inotable(inode.ino);
```
Comment (jcsp) [inline src/mds/CInode.cc]: You're returning a bool here but then ignoring it: the following save() should be conditional on the result of this

File: src/mds/InoTable.cc [hunk 2]
```diff
@@ -190,3 +190,22 @@ void InoTable::generate_test_instances(list<InoTable*>& ls)
 {
   ls.push_back(new InoTable());
 }
+
+
+bool InoTable::repair_inotable(inodeno_t id)
+{
+	dout(0) << "repair inotable " << id << dendl;
+	if (free.contains(id)) {
```
Comment (jcsp) [inline src/mds/InoTable.cc]: This check is backwards; repair needs doing if the inode number (in use by the inode we are checking) is marked as free.  We should skip if the inode is marked as *not* free (i.e. correct)

File: src/mds/InoTable.cc [hunk 3]
```diff
@@ -190,3 +190,22 @@ void InoTable::generate_test_instances(list<InoTable*>& ls)
...
+		dout(0) << "repair_inotable: Skipping ino=" << id << "pver =" << projected_version << "ver= " << version << dendl;
+		return false;
+	}
+
+	dout(0) << "repair_inotable: adding ino" << id << "to inotable" << dendl;
+	dout(0) << "repair_inotable: before status. ino=" << id << "pver =" << projected_version << "ver= " << version << dendl;
+	free.insert(id);
```
Comment (jcsp) [inline src/mds/InoTable.cc]: This is backwards, the repair action is to *remove* the inode number from the free sets

**Comment by vishalkanaujia** [comment]
@jcsp Thanks for reviewing the changes. Actually, due to a mistake, I ended up creating pull request. These changes are not at all close to completion. I will update the pull request and request you to review once it is done. But, for sure, I will take care of current review comments.

**Pull request closed without merging**